scripts/ci/do_not_merge.py

```diff
@@ -87,6 +87,37 @@ def main(argv):
         print("Pull request is description is empty.")
         fail = True
 
+    # Check if PR title is different from branch name
+    branch_name = pr.head.ref
+    pr_title = pr.title.strip()
+
+    # Convert branch name to potential default title formats that GitHub might generate
+    # GitHub typically converts branch names by replacing dashes/underscores with spaces
+    # and capitalizing the first letter
+    potential_auto_titles = [
+        branch_name,
+        branch_name.replace('-', ' '),
+        branch_name.replace('_', ' '),
+        branch_name.replace('-', ' ').replace('_', ' '),
+        branch_name.replace('-', ' ').capitalize(),
+        branch_name.replace('_', ' ').capitalize(),
+        branch_name.replace('-', ' ').replace('_', ' ').capitalize(),
+        branch_name.title(),
+        branch_name.replace('-', ' ').title(),
+        branch_name.replace('_', ' ').title(),
+        branch_name.replace('-', ' ').replace('_', ' ').title(),
+    ]
+
+    if pr_title in potential_auto_titles or pr_title.lower() in [
+        t.lower() for t in potential_auto_titles
+    ]:
```

Comment:
how about doing `lower()` on `branch_name` and `pr_title` above, then you'd get away with a single match here rather than this pycrobatics